src/Capgemini.PowerApps.PackageDeployerTemplate/Adapters/CrmServiceAdapter.cs

```diff
@@ -1,17 +1,24 @@
-﻿using System;
-using System.Collections.Generic;
-using System.IO;
-using System.Linq;
-using Microsoft.Xrm.Sdk;
+﻿using Microsoft.Xrm.Sdk;
 using Microsoft.Xrm.Sdk.Messages;
 using Microsoft.Xrm.Sdk.Query;
 using Microsoft.Xrm.Tooling.Connector;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
 
-namespace Capgemini.PowerApps.PackageDeployerTemplate.Extensions
+namespace Capgemini.PowerApps.PackageDeployerTemplate.Adapters
 {
-    public static class CrmServiceClientExtensions
+    public class CrmServiceAdapter : ICrmServiceAdapter
```

Comment:
I think this might be a little bit more cumbersome than extension methods - was the decision based on testability? It's a little bit confusing as it's not strictly an example of the adapter pattern. I think this also need to implement `IDisposable` via `Dispose()` on `CrmServiceClient`.

Reply:
This was primarily for testability, `CrmServiceClient` is sealed making testing quite difficult - I'm not sure what to call it, maybe wrapper has less meaning? I used 'Adapter' because it was the solution many threads online suggested when working with a sealed class. 

Your point on `IDisposable` is a good one - I'll add that in.

Reply:
I've implemented `IDisposable` but there's nowhere to call `Dispose()` from as the PackageDeployerTemplate isn't disposable itself. What are your thoughts?

Reply:
Actually, that's a good point. The Package Deployer will be calling `Dispose` internally on the `CrmServiceClient` instance (I would imagine) and as the `Dispose` on this would just be delegating to that, it probably doesn't make sense. Maybe if this class was being used in any other contexts.